apps/platform-app/src/app/[locale]/(profile)/(components)/VerificationModal/VerificationModal.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import { I18nNamespaces } from '@/i18n'
+import { Accordion, Button } from '@gdh/ui-system'
+import { GithubIcon } from '@gdh/ui-system/icons'
+import { PublishingState } from '@prisma/client'
+import { useTranslations } from 'next-intl'
+import styles from './VerificationModal.module.scss'
+
+interface VerificationModalProps {
+  profileStatus: PublishingState
+  rejectionReason?: string
+  onClose: () => void
+}
+
+export default async function VerificationModal({
+  profileStatus,
+  onClose,
+  rejectionReason,
+}: VerificationModalProps) {
+  const t = useTranslations(I18nNamespaces.VerificationModal)
+
+  const headerText =
+    profileStatus === PublishingState.APPROVED
+      ? t('approvedHeader')
+      : t('rejectedHeader')
+
+  const bodyText =
+    profileStatus === PublishingState.APPROVED
+      ? t('approvedText')
+      : t('rejectedText')
+
+  return (
+    <div className={styles.overlay}>
+      <div className={styles.container} data-testid="publishProfilePopup">
+        <h2 className={styles.header}>{headerText}</h2>
+        <span className={styles.text}>{bodyText}</span>
+        {profileStatus === PublishingState.REJECTED && rejectionReason && (
```

Comment:
**suggestion**: Consider using a more descriptive title for the Accordion, such as "Rejection Details".